assembl/static2/js/app/client.js

```diff
@@ -1,10 +1,25 @@
-import { ApolloClient } from 'react-apollo';
+import { ApolloClient, IntrospectionFragmentMatcher } from 'react-apollo';
 import { createNetworkInterface } from 'apollo-upload-client';
 import { getDiscussionSlug } from './utils/globalFunctions';
 import { getFullPath } from './utils/routeMap';
 import fetch from 'isomorphic-fetch'; // eslint-disable-line
 
+const myFragmentMatcher = new IntrospectionFragmentMatcher({
+  introspectionQueryResultData: {
+    __schema: {
+      types: [
+        {
+          kind: 'INTERFACE',
+          name: 'IdeaInterface',
+          possibleTypes: [{ name: 'Idea' }, { name: 'Thematic' }]
+        }
+      ]
+    }
+  }
+});
+
```

Comment:
It's documented here http://dev.apollodata.com/react/initialization.html#fragment-matcher
Apollo needs to know the schema, that Idea and Thematic ObjectTypes both implements IdeaInterface.